homeassistant/components/plex/media_player.py

```diff
@@ -20,150 +20,37 @@
-        ): vol.All(cv.time_period, cv.positive_timedelta),
-    }
-)
-
 
 def setup_platform(hass, config, add_entities_callback, discovery_info=None):
     """Set up the Plex platform."""
-    plex_data = hass.data.setdefault(PLEX_DOMAIN, {})
-    server_setup = plex_data.setdefault(SERVER_SETUP, False)
-    if server_setup:
-        return
-
-    # get config from plex.conf
-    file_config = load_json(hass.config.path(PLEX_CONFIG_FILE))
-
-    if file_config:
-        # Setup a configured PlexServer
-        host, host_config = file_config.popitem()
-        token = host_config["token"]
-        try:
-            has_ssl = host_config["ssl"]
-        except KeyError:
-            has_ssl = False
-        try:
-            verify_ssl = host_config["verify"]
-        except KeyError:
-            verify_ssl = True
-
-    # Via discovery
-    elif discovery_info is not None:
-        # Parse discovery data
-        host = discovery_info.get("host")
-        port = discovery_info.get("port")
-        host = f"{host}:{port}"
-        _LOGGER.info("Discovered PLEX server: %s", host)
-
-        if host in _CONFIGURING:
-            return
-        token = None
-        has_ssl = False
-        verify_ssl = True
-    else:
-        host = config[CONF_HOST]
-        port = config[CONF_PORT]
-        host = f"{host}:{port}"
-        token = config.get(CONF_TOKEN)
-        has_ssl = config[CONF_SSL]
-        verify_ssl = config[CONF_VERIFY_SSL]
-
-    setup_plexserver(
-        host, token, has_ssl, verify_ssl, hass, config, add_entities_callback
-    )
-
-
-def setup_plexserver(
-    host, token, has_ssl, verify_ssl, hass, config, add_entities_callback
-):
-    """Set up a plexserver based on host parameter."""
-    import plexapi.exceptions
-
-    http_prefix = "https" if has_ssl else "http"
-
-    server_config = {
-        CONF_URL: f"{http_prefix}://{host}",
-        CONF_TOKEN: token,
-        CONF_VERIFY_SSL: verify_ssl,
-    }
-
-    try:
-        plexserver = PlexServer(server_config)
-        plexserver.connect()
-    except (
-        plexapi.exceptions.BadRequest,
-        plexapi.exceptions.Unauthorized,
-        plexapi.exceptions.NotFound,
-    ) as error:
-        _LOGGER.info(error)
-        # No token or wrong token
-        request_configuration(host, hass, config, add_entities_callback)
-        return
-    else:
-        hass.data[PLEX_DOMAIN][SERVER_SETUP] = True
-
-    # If we came here and configuring this host, mark as done
-    if host in _CONFIGURING:
-        request_id = _CONFIGURING.pop(host)
-        configurator = hass.components.configurator
-        configurator.request_done(request_id)
-        _LOGGER.info("Discovery configuration done")
-
-    # Save config
-    save_json(
-        hass.config.path(PLEX_CONFIG_FILE),
-        {host: {"token": token, "ssl": has_ssl, "verify": verify_ssl}},
-    )
+    server_id = hass.data[PLEX_DOMAIN][SHARED_SERVER]
```

Comment:
Add a guard clause here that checks if discovery_info is None and return if so. This will guard against legacy stale platform config sections setting up the platform.